crates/typst/src/eval/dict.rs

```diff
@@ -167,6 +167,19 @@ impl Dict {
         Arc::make_mut(&mut self.0).insert(key, value);
     }
 
+    /// Gets a value from the dictionary.
+    /// If the key does not exist, inserts the given value and returns it.
+    #[func]
+    pub fn at_or_insert(
+        &mut self,
+        /// The key at which to retrieve the item.
+        key: Str,
+        /// The value to insert if the key is not part of the dictionary.
+        value: Value,
+    ) -> Value {
+        Arc::make_mut(&mut self.0).entry(key).or_insert(value).clone()
```

Comment:
Unconditionally making the `Arc` mutable might have more cost than `entry` is saving over double-lookup. Not sure.